subsys/bluetooth/mesh/sensor_types.c

```diff
@@ -180,6 +187,46 @@ struct scalar_repr {
 	int64_t value;
 };
 
+static int64_t scalar_decode_raw(const struct bt_mesh_sensor_format *format,
+				 const uint8_t *src)
+{
+	const struct scalar_repr *repr = format->user_data;
+
+	int64_t raw;
+
+	switch (format->size) {
+	case 1:
+		if (repr->flags & SIGNED) {
+			raw = (int8_t) *src;
+		} else {
+			raw = *src;
+		}
+		break;
+	case 2:
+		if (repr->flags & SIGNED) {
+			raw = (int16_t) sys_get_le16(src);
```

Comment:
I've been a bit on the fence about whether to do endianness handling when constructing the `bt_mesh_sensor_value` in the message handlers, or to do it only when we actually need to use the raw value for something and need to convert it to e.g. integer or float. Currently I do the latter.

I made this choice under the assumption that no one will use the raw data array directly, and will always use the API functions for e.g. converting to float if needed (at which point the endianness handling will happen). Is this a valid assumption? Or should I store the `raw` data in system endianness?